src/routes/api/users.ts

```diff
@@ -1,15 +1,66 @@
 import express from "express";
 import authWithAcessMiddleware from "../../middleware/auth/user/authWithAcessMiddleware";
 import authDataOnlyMiddleware from "../../middleware/auth/user/authDataOnlyMiddleware";
+import { UserModel } from "../../models/user";
+import bcrypt from "bcrypt";
 
 const userRouter = express.Router();
 
-userRouter.get("/:userId", authDataOnlyMiddleware, (req, res) => {
-  res.send("a specific user");
+userRouter.get("/:userId", authDataOnlyMiddleware, async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    //Find the user in db
+    const userDoc = await UserModel.findById(userId);
+    if (!userDoc) return res.status(404).json({ error: "user doesn't exist" });
+    if (userDoc.private_profile) res.status(200).json({ user: "This user profile is public" });
+
+    res.status(200).json({
+      id: userDoc.id,
+      username: userDoc.username,
+      liked_projects: userDoc.liked_projects,
+    });
+  } catch (err) {
+    console.log(err);
```

Comment:
Better to send an error to express (`next(error)`). You can use this to generate nice errors: https://www.npmjs.com/package/http-errors

Here it's also worse than before since you don't even res anything in case of error. Meaning if something goes wrong the request will never terminate.